### PR TITLE
packages: add ns-netmap

### PR DESCRIPTION
**Pull request by gsanchietti**
This is a cli-only implementation of netmap.

See [README](https://github.com/NethServer/nethsecurity/blob/netmap/packages/ns-netmap/README.md) for more info on how to use it.

TODO:
- [x] verify the behavior
- [x] implement a pre-commit hook to generate the rules

Card: https://trello.com/c/tavYHA5b/283-nat-rules-source-and-destination-netmap

### CODE REVIEW

File: packages/ns-netmap/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+- `device_in`: list of coming network interfaces (optional)
+- `device_out`: list of outgoing network interfaces (optional)
+- `map_from`: IPv4/IPv6 network address for source address translation
+- `map_to`:	IPv4/IPv6 network address for destination address translation
+
+Example of configuration file:
+```
+config rule
+	option name 'source_nat1'
+	option src '10.50.50.0/24'
```
Comment (cotosso) [inline packages/ns-netmap/README.md:19]:
```suggestion
	option dest '10.50.50.0/24'
```

File: packages/ns-netmap/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	list device_in 'eth0'
+	list device_in 'eth1'
+	list device_out 'tunrw'
+	option map_from '192.168.1.0/24'
+	option map_to '192.168.57.0/24'
+
+config rule
+	option name 'dest_nat1'
+	option dest '10.50.50.0/24'
```
Comment (cotosso) [inline packages/ns-netmap/README.md:28]:
```suggestion
	option src '10.50.50.0/24'
```

File: packages/ns-netmap/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# ns-netmap
+
+This package implements nft netmap rules thar not supported by fw4.
```
Comment (filippocarletti) [inline packages/ns-netmap/README.md:3]:
```suggestion
This package implements nft netmap rules not supported by fw4.
```

File: packages/ns-netmap/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+The configuration is saved inside `/etc/config/netmap`.
+Each record must be of type `rule`.
+Each rule can contains these fields:
+- `name`: name identifying the rule, it must respect nft limitation for comments
+- `src` (or `dest`): IPv4/IPv6 network address for source (or destination) NAT rule
+- `device_in`: list of coming network interfaces (optional)
```
Comment (filippocarletti) [inline packages/ns-netmap/README.md:10]:
```suggestion
- `device_in`: list of incoming network interfaces (optional)
```

File: packages/ns-netmap/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	option map_to '192.168.57.0/24'
+
+config rule
+	option name 'dest_nat1'
+	option src '10.50.50.0/24'
+	option map_from '192.168.1.0/24'
+	option map_to '192.168.57.0/24'
+```
+
+After adding a rule, execute the following to generate the rules:
```
Comment (filippocarletti) [inline packages/ns-netmap/README.md:33]:
```suggestion
After adding a rule, execute the following commands to generate the rules:
```

File: packages/ns-netmap/Makefile [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+PKG_MAINTAINER:=Giacomo Sanchietti <giacomo.sanchietti@nethesis.it>
+PKG_LICENSE:=GPL-3.0-only
+
+include $(INCLUDE_DIR)/package.mk
+ 
+define Package/ns-netmap
+	SECTION:=base
+	CATEGORY:=NethSecurity
+	TITLE:=Netmap firewall rules
+	URL:=https://github.com/NethServer/nethsecurity-controller/
```
Comment (filippocarletti) [inline packages/ns-netmap/Makefile:23]: Maybe https://github.com/NethServer/nethsecurity

**Review by filippocarletti — CHANGES_REQUESTED** [review]

**Review by filippocarletti — APPROVED** [review]

**Pull request merged**